internal/renderer/service_util.go

```diff
@@ -381,6 +388,57 @@ func createLbService4Gateway(c *RenderContext, gw *gwapiv1a2.Gateway) *corev1.Se
 	return svc
 }
 
+func setHealthCheck(annotations map[string]string, svc *corev1.Service) (int32, error) {
+	var healthCheckPort int32
+	var healthCheckProtocol string
+
+	// Find health check port
+	for annotationKey, annotation := range annotations {
+		if annotationRegexPort.MatchString(annotationKey) {
+			p, err := strconv.ParseInt(annotation, 10, 32)
+			if err != nil {
+				return 0, err
+			} else {
+				healthCheckPort = int32(p)
+			}
+		}
+	}
+
+	// Find health check protocol
+	for annotationKey, annotation := range annotations {
+		if annotationRegexProtocol.MatchString(annotationKey) {
+			switch strings.ToUpper(annotation) {
+			case "TCP", "HTTP":
+				healthCheckProtocol = "TCP"
+			default:
+				return 0, errors.New("unknown health check protocol")
+			}
+		}
+	}
+
+	if healthCheckPort > 0 && healthCheckProtocol != "" {
+		svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{
+			Name:     "gateway-health-check",
+			Protocol: corev1.Protocol(healthCheckProtocol),
+			Port:     int32(healthCheckPort),
+		})
+		return int32(healthCheckPort), nil
+	}
+	return 0, errors.New("missing health check port and/or protocol")
```

Comment:
I don't think this is an error per se. We definitely don't want to report "could not set health-check port" every time we find *no* DO/AWS health-check annotations in a Gateway. Just return this and let's move on:
```
return 0, nil
```